v2/pkg/astnormalization/variables_mapper_test.go

```diff
@@ -0,0 +1,432 @@
+					object(id: $a) {
+						echo(value: $b)
+						echoTwo(value: $c)
+					}
+				}`,
+
+			variablesMapping: map[string]string{
+				"a": "a",
+				"b": "b",
+				"c": "c",
+			},
+		},
+		{
+			name: "4 Inline variable and external variable",
+			input: `
+				query MyQuery($varOne: ID!) {
+					object(id: $varOne) {
+						name
+						echo(value: "Hello World!")
+					}
+				}`,
+			output: `
+				query MyQuery($a: ID! $b: String!) {
+					object(id: $a) {
+						name
+						echo(value: $b)
+					}
+				}`,
+
+			variablesMapping: map[string]string{
+				"a": "varOne",
+				"b": "a",
+			},
+		},
+		{
+			name: "5.1 Multiple external variables",
+			input: `
+				query MyQuery($varOne: ID! $varTwo: String!) {
+					object(id: $varOne) {
+						name
+						echo(value: $varTwo)
+					}
+				}`,
+			output: `
+				query MyQuery($a: ID! $b: String!) {
+					object(id: $a) {
+						name
+						echo(value: $b)
+					}
+				}`,
+			variablesMapping: map[string]string{
+				"a": "varOne",
+				"b": "varTwo",
+			},
+		},
+		{
+			name: "6 Multiple colliding external variables",
+			input: `
+				query MyQuery($a: String! $b: ID!) {
+					object(id: $b) {
+						echo(value: $a)
+						name
+					}
+				}`,
+			output: `
+				query MyQuery($a: ID! $b: String!) {
+					object(id: $a) {
+						echo(value: $b)
+						name
+					}
+				}`,
+			variablesMapping: map[string]string{
+				"a": "b",
+				"b": "a",
+			},
+		},
+		{
+			name: "7 multiple inline variables",
+			input: `
+				query MyQuery {
+					object(id: "abc123") {
+						echo(value: "Hello World!")
+						name
+					}
+				}`,
+			output: `
+				query MyQuery($a: ID! $b: String!) {
+					object(id: $a) {
+						echo(value: $b)
+						name
+					}
+				}`,
+			variablesMapping: map[string]string{
+				"a": "a",
+				"b": "b",
+			},
+		},
+		{
+			name: "8 Inline variable with multiple colliding external variables",
+			input: `
+				query MyQuery($a: ID! $b: String! ) {
+					object(id: $a) {
+						name
+						copy(input: { id: "abc123", name: "MyObject"}),
+						echo(value: $b)
+					}
+				}`,
+			output: `
+				query MyQuery($a: ID! $b: InputObject! $c: String!) {
+					object(id: $a) {
+						name
+						copy(input: $b),
+						echo(value: $c)
+					}
+				}`,
+			variablesMapping: map[string]string{
+				"a": "a",
+				"b": "c",
+				"c": "b",
+			},
+		},
+		{
+			name: "9 Inline variable with multiple colliding external variables",
+			input: `
+				query MyQuery($a: ID! $b: String! $c: String! ) {
+					object(id: $a) {
+						name
+						copy(input: { id: "abc123", name: $b})
+						echo(value: $c)
+					}
+				}`,
+			output: `
+				query MyQuery($a: ID! $b: InputObject! $c: String! ) {
+					object(id: $a) {
+						name
+						copy(input: $b)
+						echo(value: $c)
+					}
+				}`,
+			variablesMapping: map[string]string{
+				"a": "a",
+				"b": "d",
+				"c": "c",
+			},
+		},
+		{
+			name: "11 Reused external variable",
```

Comment:
Where is 10? :D

Reply:
10 was fully equal to nine, so this one could be 10